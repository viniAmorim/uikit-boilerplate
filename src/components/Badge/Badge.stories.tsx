import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Badge, BadgeProps } from './Badge';

export default {
  title: 'Atoms/Badge',
  component: Badge,
} as Meta;

const Template: Story<BadgeProps> = (args) => <Badge {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'INATIVO',
};

export const Active = Template.bind({});
Active.args = {
  children: 'ATIVO',
  badgeColor: 'green',
};

export const Waiting = Template.bind({});
Waiting.args = {
  children: 'AGUARDANDO',
  badgeColor: 'yellow',
};
