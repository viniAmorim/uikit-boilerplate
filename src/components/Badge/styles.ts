import styled, { css } from 'styled-components';

export type StyledBadgeProps = {
  badgeColor?: 'green' | 'yellow' | 'red';
};

const colorModifier = {
  green: css`
    color: ${({ theme }) => theme.colors.green700};
    background: ${({ theme }) => theme.colors.green100};
  `,
  yellow: css`
    color: ${({ theme }) => theme.colors.yellow500};
    background: ${({ theme }) => theme.colors.yellow100};
  `,
  red: css`
    color: ${({ theme }) => theme.colors.red500};
    background: ${({ theme }) => theme.colors.red100};
  `,
};

export const Badge = styled.span<StyledBadgeProps>`
  font: 700 11px/17px 'Montserrat';
  display: inline;
  justify-content: center;
  align-items: center;
  padding: 4px 6px;
  border-radius: 4px;

  ${({ badgeColor = 'red' }) => colorModifier[badgeColor]}
`;
