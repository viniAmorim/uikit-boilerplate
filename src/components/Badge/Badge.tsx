import React, { forwardRef, ReactNode } from 'react';

import * as Styled from './styles';

export interface BadgeProps
  extends React.HTMLAttributes<HTMLSpanElement>,
    Partial<Styled.StyledBadgeProps> {
  children: ReactNode;
}

export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
  { children, badgeColor = 'red', ...rest },
  ref,
) {
  return (
    <Styled.Badge ref={ref} badgeColor={badgeColor} {...rest}>
      {children}
    </Styled.Badge>
  );
});

export default Badge;
