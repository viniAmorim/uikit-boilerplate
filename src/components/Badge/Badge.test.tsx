import React from 'react';

import { render } from '~/test/utils';

import { Badge } from './Badge';

describe('Badge', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Badge>Badge</Badge>);

    const badgeElement = getByText('Badge');

    expect(badgeElement).toBeInTheDocument();
    expect(badgeElement).toHaveTextContent(/^Badge$/);

    expect(badgeElement).toMatchSnapshot();
  });

  it('should render a green badge', () => {
    const { getByText } = render(<Badge badgeColor="green">Badge</Badge>);

    const badgeElement = getByText('Badge');

    expect(badgeElement).toBeInTheDocument();

    expect(badgeElement).toMatchSnapshot();
  });

  it('should render a yellow badge', () => {
    const { getByText } = render(<Badge badgeColor="yellow">Badge</Badge>);

    const badgeElement = getByText('Badge');

    expect(badgeElement).toBeInTheDocument();

    expect(badgeElement).toMatchSnapshot();
  });
});
