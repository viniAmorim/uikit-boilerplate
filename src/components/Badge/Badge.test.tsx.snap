// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render a green badge 1`] = `
<span
  class="sc-beqWaB haAcKM"
>
  Badge
</span>
`;

exports[`Badge should render a yellow badge 1`] = `
<span
  class="sc-beqWaB DAiEW"
>
  Badge
</span>
`;

exports[`Badge should render correctly 1`] = `
<span
  class="sc-beqWaB hSpapK"
>
  Badge
</span>
`;
